src/index.ts

```diff
@@ -100,11 +66,16 @@ interface CookieListItem {
   sameSite: CookieSameSite;
 }
 
+type DeletedCookieListItem = CookieListItem & {
+  value: undefined;
+};
+
```

Comment:
This bit gives me some cause for concern. I think we have the modelling wrong for `CookieListItem`.

The [WebIDL spec states dictionary members are all optional unless specified as `required`](https://heycam.github.io/webidl/#idl-dictionaries) so I think `CookieListItem` needs to be re-typed as:

```js
interface CookieListItem {
  name?: string;
  value?: string;
  domain?: string;
  path?: string;
  expires?: number;
  secure?: boolean;
  sameSite?: CookieSameSite;
}
```

Sure enough if I use the tool [`webidl2ts`](https://github.com/giniedp/webidl2ts) I get that definition. It might be useful to double check all dictionaries we have ported to ensure they all follow this pattern.